whois-update/src/main/java/net/ripe/db/whois/update/authentication/credential/ClientCertificateCredentialValidator.java

```diff
@@ -0,0 +1,102 @@
+package net.ripe.db.whois.update.authentication.credential;
+
+import net.ripe.db.whois.common.DateTimeProvider;
+import net.ripe.db.whois.common.dao.RpslObjectDao;
+import net.ripe.db.whois.common.rpsl.ObjectType;
+import net.ripe.db.whois.common.rpsl.RpslObject;
+import net.ripe.db.whois.update.domain.ClientCertificateCredential;
+import net.ripe.db.whois.update.domain.PreparedUpdate;
+import net.ripe.db.whois.update.domain.UpdateContext;
+import net.ripe.db.whois.update.domain.UpdateMessages;
+import net.ripe.db.whois.update.domain.X509Credential;
+import net.ripe.db.whois.update.keycert.X509CertificateWrapper;
+import net.ripe.db.whois.update.log.LoggerContext;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.CheckForNull;
+import java.util.Collection;
+
+@Component
+public class ClientCertificateCredentialValidator implements CredentialValidator<ClientCertificateCredential, X509Credential> {
+
+    private final RpslObjectDao rpslObjectDao;
+    private final DateTimeProvider dateTimeProvider;
+    private final LoggerContext loggerContext;
+
+    @Autowired
+    public ClientCertificateCredentialValidator(final RpslObjectDao rpslObjectDao, final DateTimeProvider dateTimeProvider, final LoggerContext loggerContext) {
+        this.rpslObjectDao = rpslObjectDao;
+        this.dateTimeProvider = dateTimeProvider;
+        this.loggerContext = loggerContext;
+    }
+
+    @Override
+    public Class<X509Credential> getSupportedCredentials() {
+        return X509Credential.class;
+    }
+
+    @Override
+    public Class<ClientCertificateCredential> getSupportedOfferedCredentialType() {
+        return ClientCertificateCredential.class;
+    }
+
+    @Override
+    public boolean hasValidCredential(final PreparedUpdate update, final UpdateContext updateContext, final Collection<ClientCertificateCredential> offeredCredentials, final X509Credential knownCredential) {
+        for (final ClientCertificateCredential offeredCredential : offeredCredentials) {
+
+            if (verifyClientCertificate(update, updateContext, offeredCredential, knownCredential)) {
```

Comment:
Also log the offered credential in the audit log here?

Reply:
Done.